controller/webutil/gen/gen.go

```diff
@@ -0,0 +1,34 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/filecoin-project/dealbot/controller/webutil"
+)
+
+func main() {
+	if len(os.Args) < 2 {
+		fmt.Printf("Must specify source directory")
+		os.Exit(1)
+	}
+
+	cmd := exec.Command("npm", "install")
+	cmd.Dir = os.Args[1]
+	err := cmd.Run()
+	if err != nil {
+		fmt.Printf("Failed to install frontend dependencies: %v\n", err)
+		if _, err := os.Stat(path.Join(os.Args[1], "node_modules")); os.IsNotExist(err) {
+			os.Exit(1)
+		}
+	}
+
+	data := webutil.Compile(os.Args[1], true)
```

Comment:
I wonder if it might be easier to just plop a `gen.go` file in the output directory that has:

```
//go:generate npm install
//go:generate go run github.com/evanw/esbuild/cmd/esbuild [whatever args]
```

This is assuming that what we're doing with esbuild could also be done via the command line. Since we're requiring npm to be installed anyway, I don't think calling it via os/exec gains us that much. I'll ultimately defer to you though, as I've never used esbuild.

Reply:
This setup lets me run a development version of the controller that builds the frontend code from source every time it serves it, rather than using the pre-built version.

I can finish putting that in place so it's clear why i'm structuring it this way

Reply:
Ah I see, makes sense.